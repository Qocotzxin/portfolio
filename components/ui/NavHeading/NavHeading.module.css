.lastName {
    transform: translateX(8px);
    display: inline-block;
    background-clip: text;
    background-image: linear-gradient(to right, var(--chakra-colors-teal-300), var(--chakra-colors-blue-400));
    animation: move-to-r 1s linear;
}

.firstName::after {
    content: 'ristian';

}

.lastName::after {
    content: 'tchebarne';
}

.firstName::after,
.lastName::after {
    background-clip: text;
    background-image: linear-gradient(to right, var(--chakra-colors-teal-300), var(--chakra-colors-blue-400));
    animation: fade-in 2s ease-in;
}

@keyframes move-to-r {
    from {
        transform: translateX(-52px);
    }

    to {
        transform: translateX(8px);
    }
}

@keyframes fade-in {
    from {
        opacity: 0;
    }

    to {
        opacity: 1;
    }
}