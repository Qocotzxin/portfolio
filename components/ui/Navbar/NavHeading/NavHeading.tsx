import { Heading } from "@chakra-ui/react";
import { FC } from "react";
import styles from "./NavHeading.module.css";

const NavHeading: FC = () => (
  <Heading
    as="h1"
    position="relative"
    pl="15px"
    pt="1px"
    w="100%"
    fontSize="xl"
    fontWeight="normal"
    fontFamily="Oswald"
    bgGradient="linear(to-r, teal.300, blue.300)"
    bgClip="text"
    className={styles.heading}
  >
    <span className={styles.firstName}>C</span>
    <span className={styles.lastName}>E</span>
  </Heading>
);

export default NavHeading;
