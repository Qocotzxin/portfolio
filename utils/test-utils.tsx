import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "@context/hygraphContext";

const AllTheProviders: FC<PropsWithChildren> = ({ children }) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={{}}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
