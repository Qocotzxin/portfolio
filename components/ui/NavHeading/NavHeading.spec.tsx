import NavHeading from "./NavHeading";
import { render, screen } from "@utils/test-utils";

describe("NavHeading", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<NavHeading />);
    expect(container).toMatchSnapshot();
  });

  it("Should display the text CE initially.", () => {
    render(<NavHeading />);
    expect(screen.getByText(/C/)).toBeInTheDocument();
    expect(screen.getByText(/E/)).toBeInTheDocument();
  });
});
