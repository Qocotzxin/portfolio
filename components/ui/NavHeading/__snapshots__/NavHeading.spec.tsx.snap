// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavHeading Should match snapshot. 1`] = `
.emotion-0 {
  font-family: Staatliches;
  font-weight: var(--chakra-fontWeights-normal);
  font-size: var(--chakra-fontSizes-xl);
  line-height: 1.33;
  position: relative;
  padding-left: 15px;
  padding-top: 1px;
  width: 100%;
  background-image: linear-gradient(to right, var(--chakra-colors-teal-300), var(--chakra-colors-blue-400));
  color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
}

@media screen and (min-width: 48em) {
  .emotion-0 {
    line-height: 1.2;
  }
}

<div>
  <h1
    class="chakra-heading heading emotion-0"
  >
    <span
      class="firstName"
    >
      C
    </span>
    <span
      class="lastName"
    >
      E
    </span>
  </h1>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
